cloudinit/distros/ubuntu.py

```diff
@@ -10,10 +10,36 @@
 # This file is part of cloud-init. See LICENSE file for license information.
 
 import copy
+from contextlib import suppress
+from typing import Iterable, List
 
+from cloudinit import subp
 from cloudinit.distros import PREFERRED_NTP_CLIENTS, debian
 
 
+def get_available_apt_packages(pkglist: Iterable[str]) -> List[str]:
+    apt_packages = []
+    for pkg in pkglist:
+        output: str = subp.subp(
+            ["apt-cache", "search", "--names-only", f"^{pkg}$"]
+        ).stdout.strip()
```

Comment:
this will be slow.  for large apt caches, you're going to read and parse all the package lists (/var/lib/apt/lists) for every package in this list.  thats 70,000+ packages on a jammy cloud image. And it wont even exit 'apt-cache search' early when it finds a hit.

faster, maybe at cost of memory,  would be to get the whole list once and put it into a python set.


```python
out = subp.subp(["apt-cache", "pkgnames"])
all_packages = set(out.stdout.splitlines())
return set(pkglist) - all_packages
```